src/Text/Parsing/Parser/Combinators.purs

```diff
@@ -187,3 +188,21 @@ many1Till p end = do
   x <- p
   xs <- manyTill p end
   pure (x:xs)
+
+-- | Match and maps result of parser satisfying the specified predicate.
+-- | Allows to specify error message if predicate fails
+satisfyMap :: forall s m a b. Monad m => ParserT s m a -> (a -> Either String b) -> ParserT s m b
+satisfyMap p f = try do
+  x <- p
+  case f x of
+    Right a -> pure a
+    Left err -> fail err
```

Comment:
We could have added `Show a` constraint, used Maybe instead of Either, remove `satisfyMap'` and change this to `"Predicate is not satisfied with value: " <> show x`.

let me know if it's desired change.